plugins/modules/dynamodb_table.py

```diff
@@ -999,6 +1019,10 @@ def main():
     )
     client = module.client('dynamodb', retry_decorator=retry_decorator)
 
+    if module.params.get('table_class'):
+        if not module.botocore_at_least("1.23.18"):
+            module.fail_json(msg='botocore version >= 1.23.18 is required setting table_class')
+
```

Comment:
I guess you can do something like this if you want.

```suggestion
    if module.params.get('table_class'):
        module.require_botocore_at_least('1.23.18', reason='to set table_class')
```